app/src/main/java/com/github/se/cyrcle/MainActivity.kt

```diff
@@ -59,7 +67,8 @@ class MainActivity : ComponentActivity() {
               reviewViewModel,
               userViewModel,
               mapViewModel,
-              addressViewModel)
+              addressViewModel,
+              PermissionsManager.areLocationPermissionsGranted(this))
```

Comment:
I may have multiple clues
- Permission "asker", at least in my emulator, let me choose between *Precise* and *Approximate* location. How does that works with `areLocationPermissionsGranted` ? The name of the function does not help. It seems that the Mapbox documentation isn't helping much either
- I moved `PermissionsManager.areLocationPermissionsGranted(this)` right after the permission handler initialization (`:49`), and printing it returns me `false` during my first try.

Reply:
MapBox is managing the difference internally while we are left with only the approval or not from the user I think. 
For the second point, yes sean mentionned that it could come from the fact that it is compiled at the same time not taking into account the user choice until restart.